internal/exec/util/file.go

```diff
@@ -112,6 +114,33 @@ func RenderFile(l *log.Logger, c *resource.HttpClient, f types.File) *File {
 		return nil
 	}
 
+	if f.User.Name != "" {
+		u, err := user.Lookup(f.User.Name)
```

Comment:
Just switched it to using the C implementation, and added some C code for looking up groups by name. The tests don't pass, but they didn't pass before I started this venture either (they got skipped if they were run as non-root). Will debug them soon.